openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -47,15 +53,18 @@ def __init__(self, course_key, request, username=''):
             course_key,
         )
         self.effective_user = self.overview.effective_user
+        # We need to memoize `is_staff` _before_ we configure masquerade.
+        self.is_staff = has_access(self.effective_user, 'staff', self.overview).has_access
```

Comment:
Otherwise the instructor toolbar (at least) disappears because it doesn't think the masqueraded user is staff.